keps/sig-api-machinery/4006-transition-spdy-to-websockets/README.md

```diff
@@ -0,0 +1,1228 @@
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+Some Kubernetes clients need to communicate with the API Server using a bi-directional
+streaming protocol, instead of the standard HTTP request/response mechanism. A streaming
+protocol provides the ability read and write arbitrary data messages between the
+client and server, instead of providing a single response to a client request.
+For example, the commands `kubectl exec`, `kubectl attach`, `kubectl port-forward`,
+and `kubectl cp` all benefit from a bi-directional streaming protocol. Currently,
+the bi-directional streaming solution for these `kubectl` commands is SPDY/3.1. For
+the communication leg between `kubectl` and the API Server, this enhancement transitions
+the bi-directional streaming protocol to WebSockets from SPDY/3.1 for three `kubectl`
+commands: `exec`, `attach`, and `cp`.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+The SPDY streaming protocol has been deprecated for around eight years, and by now
```

Comment:
Nit: to make the KEP more "universal" how about change that it's been deprecated since 2015?

Reply:
Done